### PR TITLE
based on base class

### PR DESCRIPTION
**Pull request by slowmoyang**
project is based on SCRAM
add BaseAnalyser as boilerplate

### CODE REVIEW

File: analysis/.gitignore [hunk 1]
```diff
@@ -0,0 +1,248 @@
+
```
Comment (jshlee) [inline analysis/.gitignore]: why have 2 git ignore??

File: external/interface/MathUtils.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+#include <cmath>
+#include <iostream>
+
+namespace delphys {
+
+static const Double_t kPi = TMath::Pi();
+static const Double_t kTwoPi = TMath::TwoPi();
+
+inline Double_t ComputeDeltaPhi(Double_t phi1, Double_t phi2) {
```
Comment (jshlee) [inline external/interface/MathUtils.h]: cant you just use root functions of these?

File: external/src/MathUtils.cc [hunk 3]
```diff
@@ -0,0 +1 @@
+#include "delphys/external/interface/MathUtils.h"
```
Comment (jshlee) [inline external/src/MathUtils.cc:1]: if all defined in header, is this really needed?

Comment (slowmoyang) [reply external/src/MathUtils.cc:1]: I didn't commit properly..

File: external/interface/MathUtils.h (hunk 2 above)
Comment (slowmoyang) [reply external/interface/MathUtils.h]: is there root functions?
functions in ROOT::Math::VectorUtils require vector or TLorentzVector

File: analysis/.gitignore (hunk 1 above)
Comment (slowmoyang) [reply analysis/.gitignore]: sorry it was a my mistake
I removed one in delphys/analysis

**Comment by jshlee** [comment]
TVector2::Phi_mpi_pi can be used for instead of dPhi, but its ok

fix conflicts

**Comment by slowmoyang** [comment]
I cleaned up BuildFile.xml. we'll make BuildFile.xml using analysis/test/makeBuildIFile on the fly.

File: README.md [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
+=======
+
 ## How to git clone
 ```
-scram p -n delPhys CMSSW CMSSW_9_4_4
-cd delPhys/src
+scram p -n delphys CMSSW CMSSW_9_4_4
+cd delphys/src
 cmsenv
 git clone git@github.com:"your_user_name"/nano.git #"your_user_name" should be replaced by real user name
```
Comment (jshlee) [inline README.md:17]: have CPLUOS instead of your_user_name

**Pull request merged**